### PR TITLE
Add documentation for `statistics()` and `aclose()` on memory channels

### PR DESCRIPTION
**Pull request by A5rocks**
I'm pretty sure the reason Sphinx isn't showing them is because they have no docstrings. Easy fix.

Mentioned on Gitter.

### CODE REVIEW

**Comment by A5rocks** [comment]
pre-commit.ci autofix

File: src/trio/_channel.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def __init__(self, max_buffer_size: int | float):  # noqa: PYI041
 
 
 @attrs.frozen
-class MemoryChannelStats:
+class MemoryChannelStatistics:
```
Comment (A5rocks) [inline src/trio/_channel.py:114]: I *think* this is backwards compatible as far as we care.

Comment (CoolCat467) [reply src/trio/_channel.py:114]: If it wasn't publicaly re-exported originally, I would think this should count as backwards compatible, and if anyone does have issues we can tell them its their fault for digging into Trio's internals.

**Review by CoolCat467 — APPROVED** [review]
Looks good to me, don't foresee any issues with this.

**Review by TeamSpen210 — APPROVED** [review]
Looks good.

File: src/trio/_tests/_check_type_completeness.json [hunk 2]
```diff
@@ -20,13 +20,10 @@
     "all": [
         "No docstring found for class \"trio.MemoryReceiveChannel\"",
         "No docstring found for class \"trio._channel.MemoryReceiveChannel\"",
-        "No docstring found for function \"trio._channel.MemoryReceiveChannel.statistics\"",
-        "No docstring found for class \"trio._channel.MemoryChannelStats\"",
-        "No docstring found for function \"trio._channel.MemoryReceiveChannel.aclose\"",
+        "No docstring found for class \"trio.MemoryChannelStatistics\"",
+        "No docstring found for class \"trio._channel.MemoryChannelStatistics\"",
```
Comment (jakkdl) [inline src/trio/_tests/_check_type_completeness.json:24]: I guess you could've added a docstring to the class at the same time to get rid of these as well

**Pull request merged**